### PR TITLE
[1.1.19] Tenant label adding department label management

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

Tenant label adding department label management

### Related issues/PRs

Related issues: 

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-dist/package/db/linkis_ddl.sql [hunk 1]
```diff
@@ -1029,3 +1029,39 @@ CREATE TABLE `linkis_cg_user_ip_config` (
...
+  `org_id` varchar(16) COMMENT '部门ID',
+  `org_name` varchar(64) COMMENT '部门名字',
+  `queue_name` varchar(64) COMMENT '默认资源队列',
+  `db_name` varchar(64) COMMENT '默认操作数据库',
+  `interface_user` varchar(64) COMMENT '接口人',
+  `is_union_analyse` varchar(64) COMMENT '是否联合分析人',
+  `create_time` varchar(64) COMMENT '用户创建时间',
+  `user_itsm_no` varchar(64) COMMENT '用户创建单号',
+  PRIMARY KEY (`user_name`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE=utf8mb4_bin COMMENT ='用户部门统计INC表';
```
Comment (aiceflower) [inline linkis-dist/package/db/linkis_ddl.sql:1050]: 排序规则，默认值，唯一校验等最好添加下

File: linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_ddl.sql [hunk 2]
```diff
@@ -30,3 +30,20 @@ ALTER TABLE linkis_ps_common_lock ADD COLUMN locker VARCHAR(255) NOT NULL COMMEN
 ALTER TABLE linkis_ps_configuration_config_key ADD column template_required tinyint(1) DEFAULT 0 COMMENT 'template required 0 none / 1 must'
 
 ALTER TABLE linkis_ps_configuration_config_value modify COLUMN  config_value varchar(500);
+
+-- ----------------------------
+-- Table structure for linkis_org_user
+-- ----------------------------
+DROP TABLE IF EXISTS `linkis_org_user`;
+CREATE TABLE `linkis_cg_tenant_department_config` (
```
Comment (aiceflower) [inline linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_ddl.sql:38]: 表名不一致

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java [hunk 3]
```diff
@@ -267,4 +265,128 @@ private void parameterVerification(TenantVo tenantVo) throws ConfigurationExcept
...
+                  paramType = "query",
+                  dataType = "HttpServletRequest",
+                  name = "req",
+                  value = ""),
+          @ApiImplicitParam(
+                  paramType = "body",
+                  dataType = "DepartmentTenantVo",
+                  name = "departmentTenantVo",
+                  value = "departmentTenantVo")
+  })
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:280]: 这段代码可以去掉

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java [hunk 4]
```diff
@@ -267,4 +265,128 @@ private void parameterVerification(TenantVo tenantVo) throws ConfigurationExcept
...
+    tenantConfigService.saveDepartmentTenant(departmentTenantVo);
+    return Message.ok();
+  }
+
+  @ApiImplicitParams({
+          @ApiImplicitParam(
+                  paramType = "query",
+                  dataType = "HttpServletRequest",
+                  name = "req",
+                  value = ""),
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:312]: 去掉

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java [hunk 5]
```diff
@@ -267,4 +265,128 @@ private void parameterVerification(TenantVo tenantVo) throws ConfigurationExcept
...
+          @ApiImplicitParam(
+                  paramType = "query",
+                  dataType = "string",
+                  name = "creator",
+                  value = "creator"),
+          @ApiImplicitParam(
+                  paramType = "query",
+                  dataType = "string",
+                  name = "tenantValue",
+                  value = "tenantValue")
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:323]: 参数与方法参数不一致，缺少 page 参数

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java [hunk 6]
```diff
@@ -267,4 +265,128 @@ private void parameterVerification(TenantVo tenantVo) throws ConfigurationExcept
...
+      @RequestParam(value = "pageSize", required = false) Integer pageSize) {
+    String userName = ModuleUserUtils.getOperationUser(req, "execute queryTenantList");
+    if (!Configuration.isAdmin(userName)) {
+      return Message.error("Failed to query-tenant-list,msg: only administrators can configure");
+    }
+    if (StringUtils.isBlank(departmentId)) departmentId = null;
+    if (StringUtils.isBlank(creator)) creator = null;
+    if (StringUtils.isBlank(tenantValue)) tenantValue = null;
+    if (null == pageNow) pageNow = 1;
+    if (null == pageSize) pageSize = 20;
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:346]: 优化代码，给默认值就行

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java [hunk 7]
```diff
@@ -178,11 +186,68 @@ public TenantVo queryTenant(String user, String creator) {
...
+  public Map<String, Object> queryDepartmentTenant(
+      String departmentId, String creator, String tenantValue, Integer pageNow, Integer pageSize) {
+    Map<String, Object> result = new HashMap<>(2);
+    List<DepartmentTenantVo> tenantVos = null;
+    if (Objects.isNull(pageNow)) {
+      pageNow = 1;
+    }
+    if (Objects.isNull(pageSize)) {
+      pageSize = 20;
+    }
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java:220]: 接口处做了判断这里是不是没必要再判断了

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java [hunk 8]
```diff
@@ -178,11 +186,68 @@ public TenantVo queryTenant(String user, String creator) {
...
+      PageHelper.clearPage();
+    }
+    PageInfo<DepartmentTenantVo> pageInfo = new PageInfo<>(tenantVos);
+    result.put("tenantList", tenantVos);
+    result.put(JobRequestConstants.TOTAL_PAGE(), pageInfo.getTotal());
+    return result;
+  }
+
+  public void deleteDepartmentTenant(Integer id) throws ConfigurationException {
+    logger.info("deleteUserIP : id:{}", id);
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java:234]: 这个日志优化下

File: linkis-public-enhancements/linkis-configuration/src/main/resources/mapper/common/DepartmentTenantMapper.xml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    <update id="updateTenant" parameterType="org.apache.linkis.configuration.entity.DepartmentTenantVo">
+        UPDATE linkis_cg_tenant_department_config
+        <trim prefix="set" suffixOverrides=",">
+            <if test="creator != null">creator = #{creator},</if>
+            <if test="departmentId != null ">department_id = #{departmentId},</if>
+            <if test="tenantValue != null">tenant_value = #{tenantValue},</if>
+            <if test="createTime != null">create_time = #{createTime},</if>
+            <if test="updateTime != null">update_time = #{updateTime},</if>
+            <if test="bussinessUser != null">bussiness_user = #{bussinessUser}</if>
+        </trim>
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-configuration/src/main/resources/mapper/common/DepartmentTenantMapper.xml:74]: 代码统一一下，字段都不用加反撇号

**Review by aiceflower — APPROVED** [review]

File: linkis-dist/package/db/linkis_ddl.sql [hunk 10]
```diff
@@ -1029,3 +1029,39 @@ CREATE TABLE `linkis_cg_user_ip_config` (
...
+
+-- ----------------------------
+-- Table structure for linkis_org_user
+-- ----------------------------
+DROP TABLE IF EXISTS `linkis_org_user`;
+CREATE TABLE `linkis_cg_tenant_department_config` (
```
[the remark this answers is not in the document]
Comment (v-kkhuang) [reply linkis-dist/package/db/linkis_ddl.sql:1056]: done

File: linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_ddl.sql (hunk 2 above)
Comment (v-kkhuang) [reply linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_ddl.sql:38]: done

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java (hunk 3 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:280]: done

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java (hunk 4 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:312]: done

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java (hunk 5 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:323]: done

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java (hunk 6 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/restful/api/TenantConfigrationRestfulApi.java:346]: done

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java (hunk 7 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java:220]: done

File: linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java (hunk 8 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/java/org/apache/linkis/configuration/service/impl/TenantConfigServiceImpl.java:234]: done

File: linkis-public-enhancements/linkis-configuration/src/main/resources/mapper/common/DepartmentTenantMapper.xml (hunk 9 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-configuration/src/main/resources/mapper/common/DepartmentTenantMapper.xml:74]: done

**Pull request closed without merging**